scipy/stats/tests/test_hypotests.py

```diff
@@ -164,6 +166,435 @@ def test_callable_cdf(self):
         assert_equal((r1.statistic, r1.pvalue), (r2.statistic, r2.pvalue))
 
 
+class TestMannWhitneyU:
+    # --- Test Input Validation ---
```

Comment:
minor point: sometimes the comments explaining what you tets are stated before the definition of the method, sometimes after. i think here you can just remove it since the method name is self-explanatory

Reply:
These comments that begin with `---` were meant to mark the beginning of groups of tests, so they always appear in-between tests: The major sections are for tests related to:
- Input Validation
- Basic Functionality
- Exact Distribution of the test statistic
- Corner Cases
- Previous Issues / Bug Fixes

I'll check that as tests have been added that they still make sense, but yeah, they were written because I thought (think) those are the major types of tests needed here.

Reply:
ok, understood.